indico_toolkit/association/splitting.py

```diff
@@ -0,0 +1,32 @@
+import re
+from copy import deepcopy
+from itertools import zip_longest
+from typing import Any, Dict, Iterator
+
+
+def split_prediction(
+    prediction: Dict[str, Any], separator: str = r"\s*\n\s*"
+) -> Iterator[Dict[str, Any]]:
+    """
+    Split a single `prediction` into mutiple predictions along a `separator` regex
+    and yield new predictions having updated text and spans.
+    """
+    texts_and_delimiters = re.split(f"({separator})", prediction["text"])
+    texts = texts_and_delimiters[::2]
```

Comment:
I'd add comment in here w/ the rationale for staggering the texts/delimiters like this w/ zip_longest